scapy/contrib/bgp.py

```diff
@@ -1,169 +1,676 @@
-#! /usr/bin/env python 
-
-# http://trac.secdev.org/scapy/ticket/162
-
+#! /usr/bin/env python
+"""BGP-4 protocol with mulit-protocol support"""
 # scapy.contrib.description = BGP
 # scapy.contrib.status = loads
 
-from scapy.packet import *
+from scapy.packet import Packet, Raw, bind_layers
 from scapy.fields import *
 from scapy.layers.inet import TCP
+from scapy.contrib.bgp_fields import *
 
+def _class_dispatcher(s, class_dict, def_class, index_from=None):
```

Comment:
Porting issue... in the pre-github version, somewhere, there was an assert
which I dropped



On 7 February 2016 at 19:12, Pierre Lalet <notifications@github.com> wrote:

> In scapy/contrib/bgp.py
> <https://github.com/secdev/scapy/pull/10#discussion_r52121441>:
>
> >
> > +def _class_dispatcher(s, class_dict, def_class, index_from=None):
>
> Why is index_from defaulting to None, while it seems to have to be a
> function?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/secdev/scapy/pull/10/files#r52121441>.
>



-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellet zu werden
Gerog Kreisler